src/eos/radiationProperties/sum.cpp

```diff
@@ -33,31 +34,102 @@ PetscErrorCode ablate::eos::radiationProperties::Sum::SumTemperatureFunction(con
     PetscFunctionReturn(0);
 }
 
+PetscErrorCode ablate::eos::radiationProperties::Sum::EmissionFunction(const PetscReal *conserved, PetscReal *property, void *ctx) {
+    PetscFunctionBeginUser;
+
+    /**
+     * Only take the first value of emission in the vector.
+     * We don't want to sum the emissions because the emission value is only dependent on the temperature, not the volume / mass fraction of the contributor.
+     */
+    auto vector = (std::vector<ThermodynamicFunction> *)ctx;
+
+    *property = 0;
+    for (const auto &[subFunction, subCtx, _] : vector[0]) {
+        PetscReal propertyTmp = 0.0;
+        PetscCall(subFunction(conserved, &propertyTmp, subCtx.get()));
+        *property += propertyTmp;
+    }
+
+    PetscFunctionReturn(0);
+}
+
+PetscErrorCode ablate::eos::radiationProperties::Sum::EmissionTemperatureFunction(const PetscReal *conserved, PetscReal temperature, PetscReal *property, void *ctx) {
+    PetscFunctionBeginUser;
+
+    /**
+     * Only take the first value of emission in the vector.
+     * We don't want to sum the emissions because the emission value is only dependent on the temperature, not the volume / mass fraction of the contributor.
+     */
+    auto vector = (std::vector<ThermodynamicTemperatureFunction> *)ctx;
+
+    *property = 0;
+    for (const auto &[subFunction, subCtx, _] : vector[0]) {
+        PetscReal propertyTmp = 0.0;
+        PetscCall(subFunction(conserved, temperature, &propertyTmp, subCtx.get()));
+        *property += propertyTmp;
+    }
+
+    PetscFunctionReturn(0);
+}
+
 ablate::eos::ThermodynamicFunction ablate::eos::radiationProperties::Sum::GetRadiationPropertiesFunction(ablate::eos::radiationProperties::RadiationProperty property,
                                                                                                          const std::vector<domain::Field> &fields) const {
-    // Create the function
-    auto contextVector = std::make_shared<std::vector<ThermodynamicFunction>>();
-    auto function = ThermodynamicFunction{.function = SumFunction, .context = contextVector};
+    switch (property) {
+        case RadiationProperty::Absorptivity: {  // Create the function
```

Comment:
If we use the same function for both, can we merge this code together?  Should we add code to make sure both property sizes are the same and set the value?